README.md

```diff
@@ -146,6 +148,86 @@ title,content
 My Title,Hello World!
 ```
 
+### Creating a custom Task
+
+TODO: Add generation instructions
+
+If you have a special use case requiring iteration over an unsupported resource, you can implement the `enumerator_builder` method instead. For example, you may need to iterate over external resources fetch from some API.
+
+This method should return an object responding to `enumerator(context:)` with an Enumerator, yielding pairs of `[item, item_cursor]`. In order for your enumerator to support resuming iteration part way through, you may use the `context.cursor`.
+
+You may optionally provide an implementation for `count`, if appropriate.
+
+```ruby
+# app/tasks/maintenance/pay_last_months_invoices_task.rb
```

Comment:
Love both of these examples, but WDYT about having a `Maintenance::ShoppingListTask` example that readers can refer to for the `Writing test for a custom Task` section? I think it would improve the understandability of the test examples to see a corresponding Task 😄

Reply:
Yeah, I was wondering the same thing after I wrote them out. I'll throw something together.

It's funny writing fake code for made up problems 🤣

Reply:
Done. I replaced `PayLastMonthsInvoicesTask` with `ShoppingListTask`, and `UnbanUsersTask` with `PurgeIngredientsTask`, which I then included tests for. I took the opportunity to describe testing the enumerator builder inline and in isolation, respectively. If it's too wordy, I figure we can collapse them using `<details />`.

I actually had to change up my hypothetical `ShoppingListTask`, as the behaviour I had been "testing" for would have been really gnarly to implement because of the nested iteration and composite cursor 😅